pkg/postage/postagecontract/contract.go

```diff
@@ -263,49 +263,49 @@ func (c *postageContract) getBalance(ctx context.Context) (*big.Int, error) {
 	return abi.ConvertType(results[0], new(big.Int)).(*big.Int), nil
 }
 
-func (c *postageContract) CreateBatch(ctx context.Context, initialBalance *big.Int, depth uint8, immutable bool, label string) ([]byte, error) {
+func (c *postageContract) CreateBatch(ctx context.Context, initialBalance *big.Int, depth uint8, immutable bool, label string) (string, []byte, error) {
 
 	if depth <= BucketDepth {
-		return nil, ErrInvalidDepth
+		return "", nil, ErrInvalidDepth
 	}
 
 	totalAmount := big.NewInt(0).Mul(initialBalance, big.NewInt(int64(1<<depth)))
 	balance, err := c.getBalance(ctx)
 	if err != nil {
-		return nil, err
+		return "", nil, err
 	}
 
 	if balance.Cmp(totalAmount) < 0 {
-		return nil, ErrInsufficientFunds
+		return "", nil, ErrInsufficientFunds
 	}
 
 	err = c.ExpireBatches(ctx)
 	if err != nil {
-		return nil, err
+		return "", nil, err
 	}
 
 	_, err = c.sendApproveTransaction(ctx, totalAmount)
 	if err != nil {
-		return nil, err
+		return "", nil, err
```

Comment:
We can also return transaction hash in case of a failed transaction for more information. I also propose to add them to debug logs for easier debugging. 
Also in case of a failed transaction it would be wise to know the Gas Price and other related information (Tip/boost percent etc) for debugging purposes.
If you look at pkg/transaction/transaction.go Line 164 ` err = t.backend.SendTransaction(ctx, signedTx)` it looks like what Abel planned was to return the tx hash in case of error and he probably may have left it in the middle (or had planned to complete it later in another PR) so you can use that to actually get the details of transaction more easily. @istae

Reply:
So we already sending an error here, which is being handled in the endpoint. I think the error is enough to describe the failure of the transaction.

Reply:
we log the transaction actually, it's done by the transaction monitor service
./pkg/transaction/monitor.go L103